lib/puppet_forge/v3/release.rb

```diff
@@ -38,6 +39,31 @@ def download(path)
         end
       end
 
+      # Uploads the tarbarll to the forge
+      #
+      # @param path [Pathname] tarball file path
+      # @return resp
+      def upload(path, api_key)
+        if api_key.nil? 
+          raise "Please provide the api_key for the forge authentication"
+	end
+        encoded_string = Base64.encode64(File.open(path, "rb").read)
+
+	resp = self.class.conn.post do |req|
+                 req.url '/v3/releases'
+                 req.headers['Content-Type']='application/json'
+                 req.headers['Authorization'] = 'Bearer '+api_key
+                 req.body = '{"file":"'+encoded_string+'"}'
+                end
+      return resp
+        rescue Faraday::ClientError => e
+          if e.response && e.response[:status] == 403
+            raise PuppetForge::ReleaseForbidden.from_response(e.response)
+          else
+            raise e
+          end
+      end
```

Comment:
So in this case would the user need to instantiate the new `Release` object first and then call "upload" on it?

I was thinking it might make more sense to make this an instance method on the `PuppetForge::V3::Module` class and have it return a new instance of the `Release` class upon success. But actually I just realized you would have the same problem with that if the module didn't exist yet.

So instead, maybe this should just be a class method on the `Release` class that you could call without setting up any objects first. And it would still return a new `Release` instance (or throw an exception on error).

What do you think about that @carabasdaniel?

Reply:
@scotje Sounds good, I'll make the modification and do a new commit where I'll also update the README with an example usage